toolchain/diagnostics/diagnostic_consumer.cpp

```diff
@@ -27,10 +27,15 @@ auto StreamDiagnosticConsumer::HandleDiagnostic(Diagnostic diagnostic) -> void {
         *stream_ << "warning: ";
         break;
       case DiagnosticLevel::Note:
+        *stream_ << "note: ";
+        break;
+      case DiagnosticLevel::Context:
         break;
     }
     *stream_ << message.format_fn(message) << "\n";
-    message.loc.FormatSnippet(*stream_);
+    if (message.level != DiagnosticLevel::Context) {
+      message.loc.FormatSnippet(*stream_);
+    }
```

Comment:
I'm OK starting here, but I think we should add a TODO or do some follow-up work as a consequence.

At the moment, the `in import` message doesn't contain the name of the imported or importing library or any other information. The name of the imported library however is visible in the snippet before this change and removed after. So it somewhat looks like a regression.

However, I don't think relying on the snippet for this information is the right call, so I do think this is the right direction. My guess is we want to revisit the contextual diagnostics to tweak their output at some point, but happy for that to be in the future.

Reply:
I think the location is still there, as part of the message prefix? It's just the snippet being elided.

Reply:
Sure then, the snippet. I find snippets useful, not sure why we'd elide them. If this is what you both want though, can you please add comments explaining the choice?

Reply:
Comment added.

Reply:
FWIW, I don't feel strongly about the snippets either way. I somewhat see both sides -- they both seem useful and a bit verbose. I guess matching Clang and GCC tilts me very, very slightly towards eliding them, but I think it'd be fine to revisit that if this is a big regression Jon. Or revisit it by exploring other ways to capture some of the value.

But maybe that's OK to fix-forward given the `note: ` churn that this PR already has?

Reply:
I'm in disagreement, but I think I'm in the minority.